client/state/account-recovery/reset/test/reducer.js

```diff
@@ -214,4 +218,76 @@ describe( '#account-recovery/reset reducer', () => {
 
 		assert.deepEqual( state.requestReset.error, mockError );
 	} );
+
+	it( 'ACCOUNT_RECOVERY_RESET_SET_VALIDATION_KEY action should set the key field', () => {
+		const key = '5201314';
+		const state = reducer( undefined, {
```

Comment:
drive by:
is there any reason you aren't using the action creator here? Then this line would be:
```js
const state = reducer( undefined, setValidationKey( key ) );
```